code/modules/clothing/under/accessories/accessory.dm

```diff
@@ -33,22 +33,35 @@
 			inv_overlay = image("icon" = icon, "icon_state" = "[tmp_icon_state]_w", dir = SOUTH)
 	if(color)
 		inv_overlay.color = color
+	if(build_from_parts)
+		inv_overlay.cut_overlays()
+		inv_overlay.add_overlay(overlay_image(icon, "[icon_state]_[worn_overlay]", flags=RESET_COLOR)) //add the overlay w/o coloration of the original sprite
 	return inv_overlay
 
 /obj/item/clothing/accessory/proc/get_mob_overlay()
+	var/I
+	if(icon_override)
+		I = icon_override
+	else if(contained_sprite)
+		I = icon
+	else
+		I = INV_ACCESSORIES_DEF_ICON
 	if(!mob_overlay)
 		var/tmp_icon_state = "[overlay_state? "[overlay_state]" : "[icon_state]"]"
 		if(icon_override)
-			if("[tmp_icon_state]_mob" in icon_states(icon_override))
+			if("[tmp_icon_state]_mob" in icon_states(I))
 				tmp_icon_state = "[tmp_icon_state]_mob"
-			mob_overlay = image("icon" = icon_override, "icon_state" = "[tmp_icon_state]")
+			mob_overlay = image("icon" = I, "icon_state" = "[tmp_icon_state]")
 		else if(contained_sprite)
 			tmp_icon_state = "[src.item_state][WORN_UNDER]"
-			mob_overlay = image("icon" = icon, "icon_state" = "[tmp_icon_state]")
+			mob_overlay = image("icon" = I, "icon_state" = "[tmp_icon_state]")
 		else
-			mob_overlay = image("icon" = INV_ACCESSORIES_DEF_ICON, "icon_state" = "[tmp_icon_state]")
+			mob_overlay = image("icon" = I, "icon_state" = "[tmp_icon_state]")
 	if(color)
 		mob_overlay.color = color
+	if(build_from_parts)
+		mob_overlay.cut_overlays()
+		mob_overlay.add_overlay(overlay_image(I, "[icon_state]_[worn_overlay]", flags=RESET_COLOR)) //add the overlay w/o coloration of the original sprite
```

Comment:
worn_overlays proc already resets color. Just use
`		var/image/worn_overlays = mob_overlay.worn_overlays(I)
		if(worn_overlays)
			result_layer.overlays.Add(worn_overlays)`

Reply:
The worn_overlays proc works on item_state (In hand sprite) which ties don't have.